### PR TITLE
Add integration test support

### PR DESCRIPTION
**Pull request by slifty**
This PR adds support for unit and integration tests -- test files need to be named `.unit.test.ts` and `.int.test.ts` going forward.

Note: I was tempted to have the filename be `integration.test.ts` to avoid shorthand / be explicit but it seems like people prefer `int` out there.

This PR currently has integration tests run in sequence until an upstream bug is patched.  I'll open an issue to track that improvement.

### CODE REVIEW

File: src/__tests__/canonicalFields.int.test.ts [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+  prepareDatabaseForCurrentWorker,
+  cleanupDatabaseForCurrentWorker,
+} from '../test/utils';
 
 const agent = request.agent(app);
 
 describe('/canonicalFields', () => {
   describe('/', () => {
     it('should return HTTP Status Code 200 OK', async () => {
+      await prepareDatabaseForCurrentWorker();
```
Comment (slifty) [inline src/__tests__/canonicalFields.int.test.ts:13]: This should probably be automatic / handled by a `beforeAll` hook for any `integration.test.ts` file.

I was hoping for a way to be more explicit about which tests require a database connection -- e.g. there might be an integration test that does NOT use the database and so spinning up a schema / migrations would be a total waste of time.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -36,16 +36,16 @@ jobs:
       - name: Run build
         run: npm run build
 
-      - name: Run migrations
-        run: npm run migrate
```
[the remark this answers is not in the document]
Comment (bickelj) [reply .github/workflows/test.yml:40]: Oh my mistake, this is the workflow, not the `package.json`. Disregard.

File: src/scripts/migrate.ts [hunk 3]
```diff
@@ -6,8 +6,7 @@ const logger = root_logger.child({ source: 'migrate' });
 logger.info('Starting migrations...');
 migrate()
   .then(() => {
-    logger.info('Migrations complete.');
-    process.exit();
+    process.stdout.write('Migrations complete.\n');
```
Comment (bickelj) [inline src/scripts/migrate.ts:9]: Should it be `logger.info`?

Comment (slifty) [reply src/scripts/migrate.ts:9]: Oh yes -- I kept the wrong line when resolving the merge conflict

File: src/test/utils.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import {
```
Comment (bickelj) [inline src/test/utils.ts:1]: Is there a better name than `utils.ts`? Perhaps `setupAndTeardown.ts`, `harnessFunctions.ts`, `databaseSetup.ts`? Something like that.

File: src/test/integrationSuiteSetup.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { db } from '../database';
+
+afterAll(async () => {
+  await db.close();
+});
```
Comment (bickelj) [inline src/test/integrationSuiteSetup.ts:5]: I am still trying to find where this fits in, but frankly I have not followed closely how any of the other pieces fit together, so I don't intend to spend too long looking.

File: jest.config.int.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+var config = require('./jest.config.base.js');
+config.testMatch = ['**/?(*.)+(int).(spec|test).[jt]s?(x)'];
+config.setupFilesAfterEnv = ["<rootDir>/src/test/integrationSuiteSetup.ts"];
```
Comment (bickelj) [inline jest.config.int.js:3]: Ah. Here is where that below file is referenced.

File: src/test/integrationSuiteSetup.ts (hunk 5 above)
Comment (bickelj) [reply src/test/integrationSuiteSetup.ts:5]: Found it above.

Comment (slifty) [reply src/test/integrationSuiteSetup.ts:5]: I can add a comment explaining what is going on for the casual observer (as you saw below, this is Jest Magic and it's not explicit).

This file is also where I'd add the automatic "db up / down before each integration test" logic.

File: src/__tests__/canonicalFields.int.test.ts (hunk 1 above)
Comment (slifty) [reply src/__tests__/canonicalFields.int.test.ts:13]: While I think this is still true -- I think we can do that as an easy refactor later in the name of enabling the baseline ability to write tests.

**Review by bickelj — APPROVED** [review]
👌

**Pull request merged**